cli/src/main.rs

```diff
@@ -289,6 +296,88 @@ fn blocking_maybe_discover_service_url(
     Ok(format!("http://{addr}:{port}"))
 }
 
+/// Pull a Wasm package from a package manager, generate its manifest, and store it.
+fn pull(identifer: String) -> Result<()> {
+    let pkg_spec = PackageSpec::try_from(identifer).unwrap();
+    log::debug!("{:#?}", pkg_spec);
+    println!(
+        "📦 Identified package {}",
+        pkg_spec.profile_url().bold().blue()
+    );
+    println!("🏷  Using module {}", pkg_spec.module.bold().blue());
+    if pkg_spec.is_binary_cached() {
+        println!(
+            "✅ Binary for {} ({}) available locally.",
+            pkg_spec.fq_name().bold().green(),
+            pkg_spec.fq_digest()
+        );
+    } else {
+        println!(
+            "⌛️ Binary for {} not available locally, downloading...",
+            pkg_spec.fq_name().blue()
+        );
+        let mod_dl = download_module(&pkg_spec);
+        match mod_dl {
+            // This means the download function did not break. It does not mean that
+            // the executable was downloaded successfully... check HTTP status code.
+            Ok(status_code) => {
+                if status_code.is_success() {
+                    println!(
+                        "✅ Downloaded {} ({}) successfully.",
+                        pkg_spec.fq_name().bold().green(),
+                        pkg_spec.fq_digest()
+                    );
+                } else if status_code.is_server_error() {
+                    println!("🛑 Server error: {}", status_code);
+                    println!("   There may be an issue with this package manager.");
+                } else if status_code.is_client_error() {
+                    println!("🛑 Client error: {}", status_code);
+                    println!("{:#?}", status_code);
+                    if status_code == 404 {
+                        println!("   Failed to download from {:?}", pkg_spec.download_urls());
+                    }
+                    println!();
+                    if pkg_spec.version == "latest" && pkg_spec.registry == PackageRegistry::Wapm {
```

Comment:
I am not happy about the lengthy special case handling and hint printing inline here. It makes this function super long and hard to read. 
What's a good way of outsourcing these hints to somewhere else? Just a helper function in `registry.rs` which matches on a keyword and prints these hints? something better?
Note: these are lazy temporary hints that could get automated away over time.